sdk/tests/conformance2/glsl3/input-with-interpotaion-as-lvalue.html

```diff
@@ -0,0 +1,99 @@
+<!--
+
+/*
+** Copyright (c) 2016 The Khronos Group Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and/or associated documentation files (the
+** "Materials"), to deal in the Materials without restriction, including
+** without limitation the rights to use, copy, modify, merge, publish,
+** distribute, sublicense, and/or sell copies of the Materials, and to
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Negative tests for the use of the shader inputs with interpolation qualifier as l-value</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<link rel="stylesheet" href="../../resources/glsl-feature-tests.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+<script src="../../js/glsl-conformance-test.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+<script type="application/javascript">
+"use strict";
+description();
```

Comment:
Can you add a comment here pointing to the spec text that's related to this behavior? As far as I can tell from a quick check, these qualifiers can't be applied to input vars in vertex shaders or output vars in fragment shaders (GLSL ES 3.0.0, section 4.3). So this rule isn't violated here.  If there is another rule, please document it here so readers in the future don't have to search it.  Also, maybe also add the angle bug link here.

Reply:
Agreed, this test looks incorrect. GLSL 3.00.6 sections 4.3.4 "Input variables", section 4.3.6 "Output Variables" and section 4.3.9 "Interpolation" all indicate that it's legal (and actually required) for the qualifiers to match on the vertex shader's outputs and fragment shader's inputs.

We've had problems in this area in the past where the "invariant" qualifier had to be erased on some platforms because of inherent mismatches between desktop GLSL and GLSL ES's shader semantics. See #1924 and #1926 . Perhaps this is the root cause of this issue?

Reply:
ESSL 3.00.6 section 4.3.4: "Variables declared as in or centroid in may not be written to during shader execution." This test is not about qualifier matching between shader stages, maybe you misunderstood the intent of the test?

Reply:
I see.  I was distracted by iterating all three interpolation qualifiers and thought this test was about that.

Then can we add here this citation from spec @Oletus quoted?

Reply:
Sorry for misunderstanding the test.

Reply:
@zhenyao  Done